cmd/yamlschema/main.go

```diff
@@ -0,0 +1,82 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/docker/cli/cli/compose/loader"
+	"github.com/spf13/cobra"
+	"github.com/xeipuuv/gojsonschema"
+)
+
+func main() {
+	var index uint
+
+	rootCmd := &cobra.Command{
+		Use:   "yamlschema <YAML file> <JSON Schema file>",
+		Short: "yamlschema validates a yaml against a json schema",
+		Args:  cobra.ExactArgs(2),
+		Run: func(cmd *cobra.Command, args []string) {
+			yaml, schema := args[0], args[1]
+			parsedYaml, err := loadYaml(yaml, index)
+			if err != nil {
+				fmt.Fprintln(os.Stderr, err)
+				os.Exit(1)
+			}
+			validateYaml(schema, parsedYaml)
+		},
+	}
+	rootCmd.Flags().UintVar(&index, "index", 0, "specify a yaml index if using a multiple document yaml file")
+
+	if err := rootCmd.Execute(); err != nil {
+		os.Exit(1)
+	}
+}
+
+func loadYaml(file string, index uint) (map[string]interface{}, error) {
+	var (
+		data []byte
+		err  error
+	)
+	if file == "-" {
+		data, err = ioutil.ReadAll(os.Stdin)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read YAML from standard input: %s", err)
+		}
+	} else {
+		data, err = ioutil.ReadFile(file)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read '%s': %s", file, err)
+		}
+	}
+	yamls := bytes.Split(data, []byte("\n---"))
+	if index >= uint(len(yamls)) {
+		return nil, fmt.Errorf("yaml index '%d' out of bounds '%d'", index, len(yamls))
+	}
+	return loader.ParseYAML(yamls[int(index)])
+}
+
+func validateYaml(schema string, yaml map[string]interface{}) {
+	if !strings.HasPrefix(schema, "http") {
+		schema = fmt.Sprintf("file://%s", schema)
+	}
+	schemaLoader := gojsonschema.NewReferenceLoader(schema)
+	dataLoader := gojsonschema.NewGoLoader(yaml)
+
+	result, err := gojsonschema.Validate(schemaLoader, dataLoader)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "failed to validate yaml: %s\n", err)
```

Comment:
return an error to cobra instead?

Reply:
I think it was my first implementation, but if you return an error during the cobra command `RunE`, cobra will print the help message....
```
$ ./bin/yamlschema toto titi
Error: test
Usage:
  yamlschema <YAML file> <JSON Schema file> [flags]

Flags:
  -h, --help         help for yamlschema
      --index uint   specify a yaml index if using a multiple document yaml file
```

Reply:
You need so set this in your root command:
```
		SilenceUsage:     true,
```
see
```
	// SilenceUsage is an option to silence usage when an error occurs.
	SilenceUsage bool
```

Reply:
Right!